### PR TITLE
docs: add BaseStrategy natspec, tweak Vault natspec

### PR DESCRIPTION
**Pull request by ShadeUndertree**

### CODE REVIEW

**Review by milkyklim — APPROVED** [review]

**Comment by lehnberg** [comment]
Great job on this @ShadeUndertree! 💙

File: contracts/BaseStrategy.sol [hunk 1]
```diff
@@ -110,23 +109,43 @@ interface StrategyAPI {
...
-/*
- * BaseStrategy implements all of the required functionality to interoperate closely
- * with the core protocol. This contract should be inherited and the abstract methods
- * implemented to adapt the strategy to the particular needs it has to create a return.
+/**
+ * @title Yearn Base Strategy
+ * @author yearn.finance
+ * @notice
+ *  BaseStrategy implements all of the required functionality to interoperate
+ *  closely with the core protocol. This contract should be inherited and the
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:117]: "interoperate closely with the Vault contract"

File: contracts/BaseStrategy.sol [hunk 2]
```diff
@@ -110,23 +109,43 @@ interface StrategyAPI {
...
     }
 
-    // Name of this contract's Strategy (Must override!)
-    // NOTE: You can use this field to manage the "version" of this strategy
-    //       e.g. `StrategySomethingOrOtherV1`. It's up to you!
+    /**
+     * @notice This Strategy's name.
+     * @dev
+     *  You can use this field to manage the "version" of this Strategy, e.g.
+     *  `StrategySomethingOrOtherV1`.
```
Comment (fubuloubu) [inline contracts/BaseStrategy.sol:146]:
```suggestion
     *  `StrategySomethingOrOtherV1`. However, "API Version" is managed by `apiVersion()` function above.
```

**Review by fubuloubu — APPROVED** [review]

**Comment by ShadeUndertree** [comment]
I rebased onto 0.2.1 changes and updated some of the NatSpec, FYI

**Review by Macarse — APPROVED** [review]

**Pull request merged**